app/models/assignment.rb

```diff
@@ -774,6 +774,15 @@ def get_num_assigned(ta_id = nil)
     end
   end
 
+  def get_num_valid
+    n = 0
+    groupings.find_each do |x|
+      !x.student_memberships.empty? && n += 1
```

Comment:
Okay, please stop using the `&& ...` pattern. In general this is not nice to read.